server/athenian/api/internal/miners/github/deployment.py

```diff
@@ -2022,3 +2024,132 @@ async def load_jira_issues_for_deployments(deployments: pd.DataFrame,
                 prs[pri] = np.sort(np.array(release_keys, dtype="S"))
 
     return issues
+
+
+async def hide_anomal_first_deployments(
+    account: int,
+    meta_ids: Sequence[int],
+    deployment_facts: pd.DataFrame,
+    pdb: Database,
+    mdb: Database,
+    threshold: float = 100.0,
+) -> None:
+    """Hide the anomal first deployments by deleting their facts from pdb."""
+    log = logging.getLogger(f"{__name__}.hide_first_deployments")
+    anomal_deploys = await _search_anomal_first_deployments(
+        meta_ids, deployment_facts, mdb, threshold,
+    )
+
+    anomal_deploys = anomal_deploys
+    if not anomal_deploys:
+        return
+
+    log.info("Clearing %d anomal first deployments", len(anomal_deploys))
+
+    stmts = []
+
+    tables = (GitHubCommitDeployment, GitHubPullRequestDeployment, GitHubReleaseDeployment)
+    for Table in tables:
+        where_clause = sa.or_(*(
```

Comment:
`OR` is almost always a bad idea in PostgreSQL, it defeats the indexes. Group by the deployment name, convert `Table.repository_full_name == value` to `Table.repository_full_name.in_(grouped_values)`

Reply:
We cannot avoid ORs here, reduced when there are many repo X deploy